src/main/java/org/broadinstitute/hellbender/utils/bwa/BwaMemIndex.java

```diff
@@ -30,38 +27,272 @@
+    public static void createIndexImageFromFastaFile(final String fasta, final String imageFile) {
+        createIndexImageFromFastaFile(fasta, imageFile, Algorithm.AUTO);
+    }
+
     /**
-     * This method creates the new, single-file bwa index image from the 5 files that the "bwa index" command produces.
+     * Creates the index image file for a reference fasta file.
+     *
+     * @param fasta the location of the targeted reference.
+     * @param imageFile the location of the new index image file.
+     * @param algo the alogirthm to use to create the index.
+     *
+     * <p>
+     *     <b><i>WARNING!</i></b>: Notice that currently this method is making JNI call that might result in an abrupt process
+     *     interruption (e.g. exit or abort system call) and so the control may never be returned.
+     * </p>
+     *
+     * @throws IllegalArgumentException if any of {@code fasta} or {@code imageFile} is {@code null}.
+     *  or it does not look like it points to a fasta formatted readable file.
+     * @throws CouldNotReadReferenceException if it doesn't seem to be possible
+     *  to read reference from the provided fasta file name.
+     *
+     * @throws CouldNotCreateIndexImageException if it doesn't seem to be possible
+     *  to create or the image file.
      */
-    public static void createIndexImage( final String refName, final String indexImageFile ) {
+    public static void createIndexImageFromFastaFile( final String fasta, final String imageFile, final Algorithm algo) {
+        assertLooksLikeFastaFile(fasta);
+        assertCanCreateOrOverwriteImageFile(imageFile);
+        if (algo == null) {
+            throw new IllegalArgumentException("the input algorithm must not be null");
+        }
+
+        final File indexPrefix = createTempIndexPrefix(fasta);
         loadNativeLibrary();
-        assertNonEmptyReadable(refName+".amb");
-        assertNonEmptyReadable(refName+".ann");
-        assertNonEmptyReadable(refName+".bwt");
-        assertNonEmptyReadable(refName+".pac");
-        assertNonEmptyReadable(refName+".sa");
-        createIndexImageFile(refName, indexImageFile);
+        createReferenceIndex(fasta, indexPrefix.getPath(), algo.toBwaName());
+        createIndexImageFile(indexPrefix.getPath(), imageFile);
+        deleteIndexFiles(indexPrefix);
     }
 
-    /** create an index from an image file.  use BwaMemIndex.createIndexImage to create one, as necessary. */
+    private static void assertCanCreateOrOverwriteImageFile(final String imageFile) {
+        if (imageFile == null) {
+            throw new IllegalArgumentException("the image file cannot be null");
+        } else {
+            final File file = new File(imageFile);
+            try {
+                if (!file.createNewFile()) {
+                    if (!file.isFile() || !file.canWrite()) {
+                        throw new CouldNotCreateIndexImageException(imageFile, "already exists as a non-regular or unwritable file");
```

Comment:
I always prefer to avoid overwriting files, but maybe a boolean to allow it will be useful for some people...